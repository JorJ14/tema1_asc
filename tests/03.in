{
    "products": {
        "id1": {
            "product_type": "Coffee",
            "name": "Indonezia",
            "acidity": 5.05,
            "roast_level": "MEDIUM",
            "price": 1
        }
    },
    "producers": [
        {
            "name": "prod1",
            "products": [
                [
                    "id1",
                    2,
                    0.37
                ]
            ],
            "republish_wait_time": 0.34
        }
    ],
    "consumers": [
        {
            "name": "cons1",
            "retry_wait_time": 0.39,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 3
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    }
                ]
            ]
        },
        {
            "name": "cons2",
            "retry_wait_time": 0.16,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    }
                ]
            ]
        },
        {
            "name": "cons3",
            "retry_wait_time": 0.17,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 4
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 1
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 1
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 1
                    }
                ]
            ]
        },
        {
            "name": "cons4",
            "retry_wait_time": 0.31,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 2
                    }
                ]
            ]
        },
        {
            "name": "cons5",
            "retry_wait_time": 0.33,
            "carts": [
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 3
                    }
                ],
                [
                    {
                        "type": "add",
                        "product": "id1",
                        "quantity": 5
                    },
                    {
                        "type": "remove",
                        "product": "id1",
                        "quantity": 5
                    }
                ]
            ]
        }
    ],
    "marketplace": {
        "queue_size_per_producer": 12
    }
}
