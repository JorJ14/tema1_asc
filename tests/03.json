{
    "products": {
        "id1": {
            "product_type": "Coffee",
            "name": "Indonezia",
            "acidity": 5.05,
            "roast_level": "MEDIUM",
            "price": 1
        }
    },
    "producers": [
        {
            "name": "prod1",
            "products": [
                [
                    "id1",
                    2,
                    0.37
                ]
            ],
            "republish_wait_time": 0.34
        }
    ],
    "consumers": [
        {
            "name": "cons1",
            "retry_wait_time": 0.39,
            "carts": [
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 3
                        }
                    ],
                    "expected_cart": {
                        "id1": 3
                    }
                },
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 5
                        }
                    ],
                    "expected_cart": {
                        "id1": 5
                    }
                }
            ]
        },
        {
            "name": "cons2",
            "retry_wait_time": 0.16,
            "carts": [
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 5
                        }
                    ],
                    "expected_cart": {
                        "id1": 5
                    }
                }
            ]
        },
        {
            "name": "cons3",
            "retry_wait_time": 0.17,
            "carts": [
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 4
                        },
                        {
                            "type": "remove",
                            "product": "id1",
                            "quantity": 1
                        }
                    ],
                    "expected_cart": {
                        "id1": 3
                    }
                },
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 1
                        },
                        {
                            "type": "remove",
                            "product": "id1",
                            "quantity": 1
                        }
                    ],
                    "expected_cart": {}
                }
            ]
        },
        {
            "name": "cons4",
            "retry_wait_time": 0.31,
            "carts": [
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 2
                        }
                    ],
                    "expected_cart": {
                        "id1": 2
                    }
                }
            ]
        },
        {
            "name": "cons5",
            "retry_wait_time": 0.33,
            "carts": [
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 5
                        },
                        {
                            "type": "remove",
                            "product": "id1",
                            "quantity": 3
                        }
                    ],
                    "expected_cart": {
                        "id1": 2
                    }
                },
                {
                    "ops": [
                        {
                            "type": "add",
                            "product": "id1",
                            "quantity": 5
                        },
                        {
                            "type": "remove",
                            "product": "id1",
                            "quantity": 5
                        }
                    ],
                    "expected_cart": {}
                }
            ]
        }
    ],
    "marketplace": {
        "queue_size_per_producer": 12
    }
}
