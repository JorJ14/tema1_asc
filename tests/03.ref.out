cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons1 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons2 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons2 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons2 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons2 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons2 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons3 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons3 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons3 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons4 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons4 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons5 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
cons5 bought Coffee(name='Indonezia', price=1, acidity=5.05, roast_level='MEDIUM')
